src/draw/sw/lv_draw_sw_rect.c

```diff
@@ -402,6 +600,43 @@ LV_ATTRIBUTE_FAST_MEM static inline lv_color_t grad_get(const lv_draw_rect_dsc_t
     return lv_color_mix(dsc->bg_grad_color, dsc->bg_color, mix);
 }
 
+/*Simple mathematic trick to avoid using a divide or multiplier */
+LV_ATTRIBUTE_FAST_MEM static inline uint8_t divBy255(uint16_t a)
```

Comment:
We have `LV_UDIV255`.

Reply:
Not the same. UDIV255 multiplies and shift (but works for any number). Here, I know that the product will always be < 65535, so I can use a shift + add solution only (see comment). Not sure it'll make a significant difference although, if the gradient map is cached.